modules/api.py

```diff
@@ -615,6 +616,26 @@ async def full_check(game: Game, version: str):
             else:
                 version = "N/A"
 
+        if game.developer != developer:
+            game.add_timeline_event(TimelineEventType.ChangedDeveloper, game.developer, developer)
+
+        if game.type != type:
+            game.add_timeline_event(TimelineEventType.ChangedType, game.type.name, type.name)
+
+        if game.tags != tags:
+            if len(game.tags) < len(tags):
+                difference = [tag.text for tag in tags if tag not in game.tags]
+                game.add_timeline_event(TimelineEventType.TagsAdded, ", ".join(difference))
+            else:
+                difference = [tag.text for tag in game.tags if tag not in tags]
+                game.add_timeline_event(TimelineEventType.TagsRemoved, ", ".join(difference))
```

Comment:
tags could be both added and removed in one pass... checking just for different count isnt reliable. the code you have for differences is fine, just need to always check the differences, and then put the if based on length of differences. and not in an else case, just one after the other